pkg/client/s3client.go

```diff
@@ -127,14 +130,24 @@ func (c *S3Client) CreateBucket(bucketName string) error {
 
 //To upload a object to S3 bucket
 func (c *S3Client) UploadObject(object, bucketName string) error {
-	klog.Infof("uploading the object %s\n", object)
+	klog.Infof("uploading the file %s\n", object)
 	//Read the content of the object
 	file, err := os.Open(object)
 	if err != nil {
 		return fmt.Errorf("err opening file %s: %s", object, err)
 	}
 	defer file.Close()
 
+	//check for object in the bucket
+	regxComp := regexp.MustCompile(`(?P<FileName>[a-zA-Z0-9\_\-\(\)\.\/]+)(?P<Ext>.tar.gz|.tar|.ova|.ova.gz|.gz)$`)
+	fileExt := regxComp.FindStringSubmatch(object)[2]
+	if objectExists := c.CheckIfObjectExists(bucketName, filepath.Base(object)); objectExists {
+		klog.Infof("Object with %s already exists in the bucket %s\n", filepath.Base(object), bucketName)
+		const layout = "02012006150405"
+		object = strings.TrimSuffix(object, fileExt) + time.Now().Format(layout) + fileExt
+		klog.Infof("object will be uploaded to s3 as %s\n", filepath.Base(object))
+	}
+
```

Comment:
This check should be just removed from the client put it in the program who is writing the code.

IMO: Let's keep it simple and let the script be failed with a meaningful name, let the user decide the object name wants to upload.

Reply:
Because this becomes really difficult to predict the object name post running this command if you want to put everything in a script.

Reply:
If possible, lets another optional flag for the object-name,

if object-name != "" {
upload(file, object-name, bucket)
}else {
upload(file, filepath.base(file), bucket)
}